pkg/koordlet/runtimehooks/protocol/container_context.go

```diff
@@ -65,6 +65,15 @@ func (c *ContainerRequest) FromReconciler(podMeta *statesinformer.PodMeta, conta
 			break
 		}
 	}
+	for _, containerSpec := range podMeta.Pod.Spec.Containers {
+		if containerSpec.Name == containerName {
+			c.ContainerEnvs = map[string]string{}
```

Comment:
is c.ContainerEnvs  already empty?  If it's not empty,  the new map will override the existing values.